content/manager/security.md

```diff
@@ -4,436 +4,701 @@ title: Security
-curl -H 'Authentication-Token:MY_TOKEN' MANAGER_IP/blueprints
-
+{{< gsHighlight  bash >}} curl -H 'Authentication-Token:MY_TOKEN' <manager-ip-address>:<port>/api/v2/blueprints {{< /gsHighlight >}}
 
-# Behind the Scenes / Advanced
+<br>
+# Advanced
+## Internal Communication
 
-## Secured Request Flow
+Currently, communication between Cloudify agents and the manager or within the manager itself does not go through
+authentication or authorization. <br>
+Instead, these requests are sent over port 8101, used by the REST service to identify them as internal and allow them to
+proceed directly to the requested endpoint.
 
-![request-flow]({{< img "guide/request_flow.png" >}})
-
-## Internal communication between the Cloudify manager and other Cloudify components
-
-Currently, communication between the Cloudify agents and the Cloudify manager does not go through authentication - instead, REST calls from the agents to the manager are done to port 8101, which has the same general behavior as port 80, yet the REST service lets requests made to this port through without having them go through any of the security mechanisms.
-
-The usage of this port, however, is restricted to components on the same subnet as the Cloudify manager alone. This is done using specific security group rules, which are set up during bootstrap (for example, see [this rule in the Openstack manager blueprint](https://github.com/cloudify-cosmo/cloudify-manager-blueprints/blob/master/openstack/openstack-manager-blueprint.yaml#L206)). It is therefore impossible to bypass the security mechanisms from outside the manager's internal network by making REST calls directly to the 8101 port.
+The usage of this port, however, must be restricted to components on the same subnet as the Cloudify manager alone.
+This is done using specific security group rules, which are set up during bootstrap (for example, see [this rule in the
+Openstack manager blueprint]
+(https://github.com/cloudify-cosmo/cloudify-manager-blueprints/blob/3.3/openstack-manager-blueprint.yaml#L667)).
+It is therefore impossible to bypass the security mechanisms from outside the manager's internal network by making REST
+calls directly to the 8101 port.
 
 In future versions, all communications from and to the Cloudify manager will utilize the security mechanisms, including communication with any of Cloudify's internal components, at which time, the bypass port 8101 will be removed.
 
+<br>
+## Customize Security: Use Your Own Implementations
+### Implementing custom security components
+As shown if previous sections, it is fairly easy to register custom python implementation to replace the default userstore,
+token generators, authentication or authorization providers - simple set the path to the implementing class and add the
+properties required to instantiate it.
+<br>
+Implementing security components is just as easy - simply follow these APIs explained below.
+<br><br>
+#### Userstore driver implementation
```

Comment:
in Userstore and Token you say `is simply a class` and in Authentication and Authorization you say `is a class`. I think the wording should be consistent. It is mostly a matter of taste, but I would remove the `simply`